### PR TITLE
Tw 673 sort by deposited only

### PR DESCRIPTION
**Pull request by herkoss**

### CODE REVIEW

**Review by keshan3262 — APPROVED** [review]

File: src/hooks/use-filtered-farms.hook.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        return isRewardsTokenSearched || isStakedTokenSearched;
+      });
+    }
+
+    switch (sortField) {
+      case FarmsSortFieldEnum.APY:
+        result.sort((farmA, farmB) => new BigNumber(farmB?.item?.apr ?? 0).minus(farmA?.item?.apr ?? 0).toNumber());
+        break;
+      case FarmsSortFieldEnum.Newest:
+        result.reverse();
```
Comment (keshan3262) [inline src/hooks/use-filtered-farms.hook.ts:48]: If liquidity baking "farm", which is the oldest, should be at the bottom by default, then this sorting won't work correctly and we should make up another way to sort farms. Moreover, Youves farms may add some more headaches here. Consider fetching contracts origination date from TZKT

File: src/screens/earn/farm-item/farm-item.styles.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import { useButtonLargeSecondaryStyleConfig } from 'src/components/button/button-large/button-large-secondary/button-large-secondary.styles';
 import { black } from 'src/config/styles';
-import { createUseStylesConfig, createUseStylesMemoized } from 'src/styles/create-use-styles';
+import { createUseStyles, createUseStylesConfig } from 'src/styles/create-use-styles';
 import { formatSize } from 'src/styles/format-size';
 import { generateShadow } from 'src/styles/generate-shadow';
 
-export const useFarmItemStyles = createUseStylesMemoized(({ colors, typography }) => ({
+export const useFarmItemStyles = createUseStyles(({ colors, typography }) => ({
```
Comment (keshan3262) [inline src/screens/earn/farm-item/farm-item.styles.ts:7]: Does `createUseStylesMemoized` cause any trouble here?

**Review by keshan3262 — APPROVED** [review]

File: src/utils/earn.utils.ts [hunk 3]
```diff
@@ -48,3 +49,14 @@ export const calculateYouvesFarmingRewards = (
...
 export const aprToApy = (aprPercentage: number, compoundFrequency = APPROXIMATE_DAYS_IN_YEAR) =>
   ((1 + Number(aprPercentage) / 100 / compoundFrequency) ** compoundFrequency - 1) * 100;
+
+export const sortByNewest = (farmA: SingleFarmResponse, farmB: SingleFarmResponse) =>
+  new Date(farmB?.item?.firstActivityTime ?? Date.now()).getTime() -
+  new Date(farmA?.item?.firstActivityTime ?? Date.now()).getTime();
+
+export const sortByOldest = (farmA: SingleFarmResponse, farmB: SingleFarmResponse) =>
+  new Date(farmA?.item?.firstActivityTime ?? Date.now()).getTime() -
+  new Date(farmB?.item?.firstActivityTime ?? Date.now()).getTime();
```
Comment (keshan3262) [inline src/utils/earn.utils.ts:59]:
```suggestion
export const sortByOldest = (farmA: SingleFarmResponse, farmB: SingleFarmResponse) => -sortByNewest(farmA, farmB);
```
Please check for prettier errors before committing this suggestion

File: src/utils/earn.utils.ts [hunk 4]
```diff
@@ -54,9 +54,7 @@ export const sortByNewest = (farmA: SingleFarmResponse, farmB: SingleFarmRespons
   new Date(farmB?.item?.firstActivityTime ?? Date.now()).getTime() -
   new Date(farmA?.item?.firstActivityTime ?? Date.now()).getTime();
 
-export const sortByOldest = (farmA: SingleFarmResponse, farmB: SingleFarmResponse) =>
-  new Date(farmA?.item?.firstActivityTime ?? Date.now()).getTime() -
-  new Date(farmB?.item?.firstActivityTime ?? Date.now()).getTime();
+export const sortByOldest = (farmA: SingleFarmResponse, farmB: SingleFarmResponse) => sortByNewest(farmB, farmA);
```
Comment (keshan3262) [inline src/utils/earn.utils.ts:57]:
```suggestion
export const sortByOldest = (farmA: SingleFarmResponse, farmB: SingleFarmResponse) => -sortByNewest(farmB, farmA);
```

**Review by keshan3262 — CHANGES_REQUESTED** [review]

**Review by keshan3262 — APPROVED** [review]

**Review by keshan3262 — APPROVED** [review]

File: src/hooks/use-filtered-farms.hook.ts [hunk 5]
```diff
@@ -33,15 +33,15 @@ export const useFilteredFarms = () => {
...
+            symbol: farm?.item?.rewardToken?.metadata?.symbol,
+            address: farm?.item?.rewardToken?.contractAddress
           },
           lowerCaseSearchValue
         );
-        const isStakedTokenSearched = farm.item.tokens.find(token =>
+        const isStakedTokenSearched = farm?.item?.tokens.find(token =>
           isAssetSearched(
-            { name: token.metadata.name, symbol: token.metadata.symbol, address: token.contractAddress },
+            { name: token?.metadata?.name, symbol: token?.metadata?.symbol, address: token?.contractAddress },
```
Comment (keshan3262) [inline src/hooks/use-filtered-farms.hook.ts:44]: The same

**Review by keshan3262 — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]
build number: 1.15_768 android done

**Review by vladvn1905 — APPROVED** [review]
ios done build: 1.15_768

**Pull request merged**